packages/block-library/src/post-author/block-colors-selector.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	Dropdown,
+	ToolbarGroup,
+	SVG,
+	Path,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { DOWN } from '@wordpress/keycodes';
+
+const ColorSelectorSVGIcon = () => (
+	<SVG xmlns="https://www.w3.org/2000/svg" viewBox="0 0 20 20">
+		<Path d="M7.434 5l3.18 9.16H8.538l-.692-2.184H4.628l-.705 2.184H2L5.18 5h2.254zm-1.13 1.904h-.115l-1.148 3.593H7.44L6.304 6.904zM14.348 7.006c1.853 0 2.9.876 2.9 2.374v4.78h-1.79v-.914h-.114c-.362.64-1.123 1.022-2.031 1.022-1.346 0-2.292-.826-2.292-2.108 0-1.27.972-2.006 2.71-2.107l1.696-.102V9.38c0-.584-.42-.914-1.18-.914-.667 0-1.112.228-1.264.647h-1.701c.12-1.295 1.307-2.107 3.066-2.107zm1.079 4.1l-1.416.09c-.793.056-1.18.342-1.18.844 0 .52.45.837 1.091.837.857 0 1.505-.545 1.505-1.256v-.515z" />
+	</SVG>
+);
+
+/**
+ * Color Selector Icon component.
+ *
+ * @param {Object} colorControlProps colorControl properties.
+ * @return {*} React Icon component.
+ */
+const ColorSelectorIcon = ( { style, className } ) => {
+	return (
+		<div className="block-library-colors-selector__icon-container">
+			<div
+				className={ `${ className } block-library-colors-selector__state-selection` }
+				style={ style }
+			>
+				<ColorSelectorSVGIcon />
+			</div>
+		</div>
+	);
+};
+
+/**
+ * Renders the Colors Selector Toolbar with the icon button.
+ *
+ * @param {Object} colorControlProps colorControl properties.
+ * @return {*} React toggle button component.
+ */
+const renderToggleComponent = ( { TextColor, BackgroundColor } ) => ( {
```

Comment:
Nit, since this is an actual component, I'd drop the `render` prefix. And maybe add something to indicate that this is about the color selector:

```suggestion
const ColorSelectorToggle = ( { TextColor, BackgroundColor } ) => ( {
```

Reply:
Oh, I just noticed https://github.com/WordPress/gutenberg/pull/19894/files#r375932904:

> I am unsure about the future of this file. It is for now copied from the nav block.

Can we import that file rather than duplicating it (and ideally extract to a different directory -- `packages/components/` comes to mind)? If we do end up duplicating, we should at least add an inline note that it's copied.

Reply:
I will try to move this to `packages/block-editor` as it is similar to `AlignmentToolbar` for example.